### ISSUE TITLE
#esoteric-python Cog

### ISSUE THREAD

**Issue by pydis-bot** [python-discord/sir-lancebot#312]
**Originally posted by Shawn McLaughlin:**

**TLDR** I believe that the #esoteric-python channel could benefit from some bot commands.

--- 

Of course, we currently only have a handful of users who actively participate in challenges every week, so the usefulness of this can be debated. I do think though that if we deem this unworthy for the moment that we should reconsider at a later date if I manage to get more people into the coding challenges. Also, having bot commands related to #esoteric-python might bring a bit more awareness to the channel, too.

## Basic functionality:

* !eso command with options;
  * suggest: Suggest a challenge for future weeks (I find myself soliciting suggestions every week and it would be neat to have a way to catalogue suggestions and build a collection for future challenges
  * vote: Eventually, the bot could perhaps generate some polls through reactions that could be pinned in the channel which would allow users to vote on which challenge seems more appealing to them
  * point system: While I don't think making esoteric-python competitive is the right path to take, I do think we could benefit from at least some recognition from people who *participate*, and could have a sort of "hall of fame" option that displays a nice richembed showing users who have contributed a lot of solutions
  * info: Explain what the heck #esoteric-python is all about, since I'm certain many people don't quite know what goes on in there and when they walk in seeing Juan's 30-deep nested lambdas skedaddle out of the channel really quickly

That being said, I think the main features for the first iteration would be the `suggest` and `info` commands. I am willing to program this cog should we deem it worthy of inclusion.

Thank you for listening

**Comment by scragly** [comment]
Transferred this issue to seasonalbot due to being an event-related feature.

**Comment by Akarys42** [comment]
I don't think this is needed anymore since we don't have eso challenges anymore. At worst we can have a tag explaining what esoteric python is.
